src/server/transaction.cc

```diff
@@ -124,13 +118,15 @@ void Transaction::InitShardData(absl::Span<const PerShardCache> shard_index, siz
     sd.arg_count = si.args.size();
     sd.arg_start = args_.size();
 
-    // Reset mask to allow locking multiple times.
-    if (incremental_locking)
-      sd.local_mask = 0;
+    if (multi_) {
+      sd.local_mask = multi_->IsIncrLocks() ? 0 : sd.local_mask & ~ACTIVE;
+    }
 
```

Comment:
Need to add comments here. Not sure about only the `sd.local_mask & ~ACTIVE`. All we need for this to work is:
- remove ACTIVE in all multi cases
- remove any LOCK flags for incremental stuff

But for "correctness" we also need to remove flags like ARMED in LOCK_AHEAD mode 🤔